extensions/Worker.Extensions.EventGrid/src/TypeConverters/EventGridBinaryDataConverter.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.EventGrid.TypeConverters
+{
+    /// <summary>
+    /// Converter to bind to BinaryData or BinaryData[] parameter.
+    /// </summary>
+    [SupportedConverterType(typeof(BinaryData))]
+    [SupportedConverterType(typeof(BinaryData[]))]
+    internal class EventGridBinaryDataConverter : IInputConverter
+    {
+        private readonly IOptions<WorkerOptions> _workerOptions;
+
+        public EventGridBinaryDataConverter(IOptions<WorkerOptions> workerOptions)
+        {
+            _workerOptions = workerOptions ?? throw new ArgumentNullException(nameof(workerOptions));
+        }
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.TargetType != typeof(BinaryData) && context.TargetType != typeof(BinaryData[]))
+            {
+                return new(ConversionResult.Unhandled());
+            }
+
+            try
+            {
+                var contextSource = context?.Source as string;
+
+                if (contextSource is null)
+                {
+                    return new(ConversionResult.Unhandled());
+                }
+
+                if (context!.TargetType == typeof(BinaryData))
+                {
+                    var binaryData = BinaryData.FromObjectAsJson<string>(contextSource);
```

Comment:
This would end up reserializing the JSON object into a JSON string, instead of wrapping the JSON object into a BinaryData instance directly (which is what we want).

```c#
var json = "{ \"name\": \"John\" }";
Console.WriteLine(BinaryData.FromObjectAsJson<string>(json));
Console.WriteLine(BinaryData.FromString(json));
```